tst/com/amazon/corretto/crypto/provider/test/EvpSignatureTest.java

```diff
@@ -84,18 +103,33 @@ public TestParams(String base, String algorithm, int length, boolean readOnly, b
             }
 
             final Signature jceSigner = getJceSigner();
+            if (paramSpec != null) {
+                jceSigner.setParameter(paramSpec);
+            }
             jceSigner.initSign(keyPair.getPrivate());
             jceSigner.update(message);
             goodSignature = jceSigner.sign();
         }
 
         @Override
         public String toString() {
-            return String.format("%s length %s. Read-only: %s, Sliced: %s",
+            final PSSParameterSpec pssParamSpec = (PSSParameterSpec) paramSpec;
+            final String pssParams = pssParamSpec == null
+                ? null
+                : String.format(
+                    "{PSS md: %s, MGF1 md: %s, saltLen: %d}",
+                    pssParamSpec.getDigestAlgorithm(),
+                    pssParamSpec.getMGFParameters() == null
+                        ? null
+                        : ((MGF1ParameterSpec) pssParamSpec.getMGFParameters()).getDigestAlgorithm(),
+                    pssParamSpec.getSaltLength()
+                );
```

Comment:
Please be cautious with the ternary operator. It's easy for it to make things really hard to read.

You don't need the second null-check at all because we can assume that the MGFParameters aren't null as that would be invalid (and we're creating them ourselves)

```suggestion
            String pssParams = null;
            if (pssParamSpec != null) {
                pssParams = String.format(
                    "{PSS md: %s, MGF1 md: %s, saltLen: %d}",
                    pssParamSpec.getDigestAlgorithm(),
                    ((MGF1ParameterSpec) pssParamSpec.getMGFParameters()).getDigestAlgorithm(),
                    pssParamSpec.getSaltLength()
                );
            }
```

Reply:
readability concerns understood, i'll update to use conventional conditional blocks. and sure, i can remove the second null check.